comms/dht/src/envelope.rs

```diff
@@ -128,14 +128,14 @@ impl DhtMessageType {
 /// It is preferable to not to expose the generated prost structs publicly.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct DhtMessageHeader {
-    pub version: u32,
+    pub major: u32,
```

Comment:
Nit discussion: at these fields maybe a bit big for the purpose? u16 should be plenty? There are just going to be a lot of headers on the network so might be worthwhile to save a few bytes here.

Reply:
**TLDR** went with the dead simple approach because the saving is probably not worth any additional code.

This is the non-generated representation of the generated protobuf struct, so I will refer

Let's compare two approaches:

Note:
-  All ints, regardless of their proto file definition, are encoded using varints. e.g a protobuf field as `uint64` with a value of 1 takes 1 byte when encoded as varint.
- There is no uint16 protobuf integer type (I would have chosen this, but I suspect it makes no practical difference)
- Byte cost is over-the-wire

Given we want to represent major = 1, minor =2 - v1.2 for short

**Approach 1**

`x: u16` with `major = msb(x) -> u8` and `minor = lsb(x) -> u8`

Byte cost:  [field number 1 byte] + [2 bytes for the u16 number with value > 127] 
                  Or as binary [0000 0001] [1000 0010]
Total: 3 bytes

Simple conversion code required to extract 2 u8s from a u16 and vice versa.

** Approach 2 **

2 fields, major and minor

Byte cost: [`major` field number 1 byte] + [value (< 127) 1 byte] + [`minor` field number 1 byte] + [value 1 byte]
Total: 4 bytes

No additional code required

Advantage: minor version can be bigger than 255, possible that we increment it with each release.

Reply:
Ja, if protobuf can't just do u16 out the box then def not worth it. I agree that u8 on minor version is not enough.

Reply:
Since it's VarInt in protobuf, it's kinda moot. Not sure how it's encoded in the SAF db, but that can be update to varint as well and is not strict for consensus